### ISSUE TITLE
Does not work with React 16

### ISSUE THREAD

**Issue by bradennapier** [wwayne/react-tooltip#342]
This package breaks any app using it once updated to React 16

```

ReactBaseClasses.js:66 Uncaught TypeError: this.updater.enqueueCallback is not a function
    at ReactTooltip.ReactComponent.setState (ReactBaseClasses.js:66)
    at resetState (index.js:436)
    at ReactTooltip.hideTooltip (index.js:448)
```

**Comment by emily-plummer** [comment]
heya @bradennapier did you end up finding a work around for this? I saw your comment in the React repo!

**Comment by bradennapier** [comment]
Hey @emily-plummer this happens when you have two versions of React installed.  This was due to a repo which I was not even including but since it was in my package.json it was causing an issue.

```
yarn list
```

revealed that repo easily and upon removing it this problem instantly resolved.  So it was not an issue with `react-tooltip` at all.  

That being said, I will leave this open for a little while so others can see it -s ince we have 7 thumbs up already I would assume quite a few are running into this!

**Comment by jordmccord** [comment]
I've been running the plugin in production for over a week now and had no problems...

**Comment by bradennapier** [comment]
Correct, you will not have problems unless there is a module that has React < 16 in its dependencies rather than peerDependencies where it should be.

**Comment by emily-plummer** [comment]
Rad! Thanks @bradennapier  :)

**Comment by alex-cory** [comment]
Wait.. so what was this fix? Moving `react` in `package.json` of ReactTooltip to `peerDependencies`??

**Comment by bradennapier** [comment]
No, it wasnt react-tooltip that was the problem, if ANY of your dependencies or their dependencies have react as a dependency rather than peerDependency then any calls that utilize enqueue callback 

```
this.setState({
    foo: 'bar'
}, () => console.log('set!'))
```

will generate this error.

if you run the 

```
yarn list 
```
 
**or**

```
npm ls 
```

on your app you should see a package that has react as a dependency -- that package needs to move it to peerDependencies or you need to remove that package.  Even if you do not use that package and it is never imported - it appears to cause a conflict.

**Comment by jonricaurte** [comment]
This line: https://github.com/wwayne/react-tooltip/blob/master/package.json#L42

Needs to change to this:

https://github.com/eiriklv/react-masonry-component/blob/master/package.json#L39

Or what this PR is doing:

https://github.com/wwayne/react-tooltip/pull/321

**Comment by wwayne** [comment]
Merged and deployed as 3.4.0
